src/clj/ataru/forms/form_payment_info.clj

```diff
@@ -0,0 +1,127 @@
+(ns ataru.forms.form-payment-info
+  (:require [ataru.middleware.user-feedback :refer [user-feedback-exception]]
+            [ataru.tarjonta-service.tarjonta-protocol :as tarjonta]
+            [clojure.string :as str]
+            [taoensso.timbre :as log]
+            [ataru.config.core :refer [config]]))
+
+(def form-payment-types
+    #{:payment-type-tutu
+      :payment-type-kk
+      :payment-type-astu
+      nil})
+
+(def kk-processing-fee
+  (bigdec (get-in config [:form-payment-info :kk-processing-fee])))
+
+(def tutu-processing-fee
+  (bigdec (get-in config [:tutkintojen-tunnustaminen :maksut :decision-amount])))
+
+(defn- requires-higher-education-application-fee?
+  [tarjonta-service haku]
+  (let [hakukohteet (tarjonta/get-hakukohteet tarjonta-service (:hakukohteet haku))]
+    (and
+      haku
+      hakukohteet
+      ; Kohdejoukko must be korkeakoulutus
+      (str/starts-with? (:kohdejoukko-uri haku) "haunkohdejoukko_12#")
+      ; "Kohdejoukon tarkenne must be empty or siirtohaku
+      (or (nil? (:kohdejoukon-tarkenne-uri haku))
+          (str/starts-with? (:kohdejoukon-tarkenne-uri haku) "haunkohdejoukontarkenne_1#"))
+      ; Must be tutkintoon johtava
+      (some true? (map #(:tutkintoon-johtava? %) hakukohteet)))))
+
+(defn- valid-fees?
+  [payment-type processing-fee decision-fee]
+  (let [fee-nonpositive? (fn [amount] (and (some? amount) (<= amount 0.00M)))
+        incorrect-kk-fee? (fn [payment-type processing-fee decision-fee]
+                            (and (= :payment-type-kk payment-type)
+                                 (or (not= processing-fee kk-processing-fee)
+                                     (some? decision-fee))))
+        incorrect-tutu-fee? (fn [payment-type processing-fee decision-fee]
+                              (and (= :payment-type-tutu payment-type)
+                                   (or (not= processing-fee tutu-processing-fee)
+                                       (some? decision-fee))))
+        incorrect-astu-fee? (fn [payment-type processing-fee _]
+                              (and (= :payment-type-astu payment-type)
+                                   (some? processing-fee)))]
+    (cond
+      (fee-nonpositive? processing-fee)
+      (do (log/warn "Nonpositive processing fee: " processing-fee)
+          false)
+
+      (fee-nonpositive? decision-fee)
+      (do (log/warn "Nonpositive decision fee: " processing-fee)
```

Comment:
varmaan pitäis logittaa mitä on tarkistettu (decision-fee)

Reply:
Hyvä huomio, fiksattu!